lib/sort_by_length.rb

```diff
@@ -1,7 +1,28 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: O(n)
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
+  array_of_words = my_sentence.split
+  sorted_array = []
+  array_of_words.each_with_index do |word, index_on_normal_array|
+    if index_on_normal_array == 0
+      sorted_array.push(array_of_words[0])
+    else
+      sorted_array_clone = sorted_array.clone
```

Comment:
With each iteration you're making a clone of `sorted_array`?  Since you're copying each element this one line is O(n) time and space complexity.